domain/tshirt.js

```diff
@@ -10,19 +10,26 @@ function invalidData(body) {
   return _.isUndefined(body.size) || _.isUndefined(body.model);
 }
 
+tshirt.NoTShirt = function () {
```

Comment:
I'm not a fan of null-objects therefore I would suggest using something like an Optional.. but in general I'm still not sold on the whole idea, of having tshirt data in the partcipants object, even if there was no tshirt ordered.